### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by acikek**

### CODE REVIEW

File: src/main/java/com/denizenscript/clientizen/events/ClientizenScriptEventRegistry.java [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package com.denizenscript.clientizen.events;
 
+import com.denizenscript.denizencore.events.ScriptEvent;
+
 public class ClientizenScriptEventRegistry {
 
 	public static void registerEvents() {
-
+		ScriptEvent.registerScriptEvent(KeyPressReleaseEvent.class);
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/events/ClientizenScriptEventRegistry.java:8]: Looks like you forgot to register `ScreenOpenCloseEvent`? or is there a reason it's not being registered?

File: src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		instance = this;
+	}
+
+	@Override
+	public boolean matches(ScriptPath path) {
+		String keyMatcher = path.eventArgLowerAt(1);
+		if (!keyMatcher.equals("key") && !runGenericCheck(keyMatcher, key.getName())) {
+			return false;
+		}
+		if (pressed != path.eventArgLowerAt(2).equals("pressed")) {
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java:30]: That check feels a bit iffy as opposed to the `thing && !arg matches thing` style checks that are used in other places, might want to change that for consistency but ig it's fine either way

File: src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		TYPE_MAP.put("options", OptionsScreen.class);
+		TYPE_MAP.put("advancements", AdvancementsScreen.class);
+	}
+
+	public String type;
+	public boolean opened;
+	public boolean switched;
+
+	public ScreenOpenCloseEvent() {
+		registerCouldMatcher("(inventory|creative|pause|options|advancements) screen opened|closed");
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java:35]: Shouldn't the screen matcher be required (I.e. no `()`)? also I think you'd have `<'inventory'>` or `<'screen'>` as opposed to listing all types, and have all possible types in the meta (which we don't have yet, ofc)

File: src/main/java/com/denizenscript/clientizen/mixin/KeyBindingMixin.java [hunk 4]
```diff
@@ -17,10 +17,10 @@ public class KeyBindingMixin {
 
 	@Inject(method = "setKeyPressed", at = @At("HEAD"))
 	private static void clientizen$onKeyPressStateChanged(InputUtil.Key key, boolean pressed, CallbackInfo ci) {
-		if (KeyPressReleaseServerEvent.instance.enabled && key.getCategory() == InputUtil.Type.KEYSYM) {
+		if (KeyPressReleaseEvent.instance != null && key.getCategory() == InputUtil.Type.KEYSYM) {
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/mixin/KeyBindingMixin.java:20]: Miiiight be wrong, but don't think it'd ever be null? Denizen-Core creates a new instance when registering it, which means the `instance` field will have a value pretty much immediately

File: src/main/java/com/denizenscript/clientizen/mixin/MinecraftClientMixin.java [hunk 5]
```diff
@@ -20,18 +20,18 @@ public class MinecraftClientMixin {
...
 	private void clientizen$screenClosed(Screen screen, CallbackInfo ci) {
 		if (canFire(currentScreen)) {
-			ScreenOpenCloseServerEvent.instance.handleScreenChange(currentScreen, screen, false);
+			ScreenOpenCloseEvent.instance.handleScreenChange(currentScreen, screen, false);
 		}
 	}
 
 	public boolean canFire(Screen screen) {
-		return screen != null && ScreenOpenCloseServerEvent.instance != null && ScreenOpenCloseServerEvent.instance.enabled;
+		return screen != null && ScreenOpenCloseEvent.instance != null;
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/mixin/MinecraftClientMixin.java:35]: Same here, pretty sure the `instance` would never be null

File: src/main/resources/clientizen.mixins.json [hunk 6]
```diff
@@ -2,13 +2,13 @@
   "required": true,
   "minVersion": "0.8",
   "package": "com.denizenscript.clientizen.mixin",
-  "compatibilityLevel": "JAVA_17",
+  "compatibilityLevel": "JAVA_17"
```
Comment (tal5) [inline src/main/resources/clientizen.mixins.json:5]: Why was the `,` removed? wouldn't that break the JSON syntax?

**Comment by tal5** [comment]
Looks like you removed all usage of `ServerEvent` stuff but not the actual classes, should delete that as well

File: src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java (hunk 2 above)
Comment (acikek) [reply src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java:30]: That would not account for the released event. If the arg is `pressed` and the key was released, the boolean `pressed` will be false, missing out on that check entirely and letting it pass.

Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/events/KeyPressReleaseEvent.java:30]: Ah right, my bad - disregard this

File: src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java (hunk 3 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java:35]: This is still unresolved

File: src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		instance = this;
+	}
+
+	@Override
+	public boolean matches(ScriptPath path) {
+		String screenMatcher = path.eventArgLowerAt(0);
+		if (!screenMatcher.equals("screen") && !runGenericCheck(screenMatcher, type)) {
+			return false;
+		}
+		if (opened && !path.eventArgLowerAt(path.eventArgs.length == 3 ? 2 : 1).equals("opens")) {
```
Comment (tal5) [inline src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java:45]: This compares it to `opens` but according to the syntax it's `opened`

File: src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java (hunk 3 above)
Comment (acikek) [reply src/main/java/com/denizenscript/clientizen/events/ScreenOpenCloseEvent.java:35]: I suppose due to the number of screens that will be matched, there's no point in having an optional.

**Review by tal5 — APPROVED** [review]

**Pull request merged**